src/components/button/index.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { FC, ReactNode } from 'react';
+import {
+  ButtonProps,
+  Button as CUButton,
+  StyleProps,
+  useTheme,
+  useToken,
+} from '@chakra-ui/react';
+
+interface Props extends ButtonProps {
+  children: ReactNode;
+  size?: 'md' | 'lg' | 'xl';
+  variant?: 'primary';
+}
+interface ButtonStates {
+  normal: StyleProps;
+  hover: StyleProps;
+  active: StyleProps;
+  disabled: StyleProps;
+}
+
+const Button: FC<Props> = ({
+  children,
+  variant = 'primary',
+  size = 'lg',
+  ...rest
+}) => {
+  const btnShadowColor = useToken('colors', 'custom.btnShadow');
+
+  const { name } = useTheme();
+
+  let buttonStates: ButtonStates;
+
+  const fontSize = {
```

Comment:
We should avoid those kinds of mappings and align the theme to make it fit, the same goes for spacing etc. Otherwise, we will need to map sizes and spacings for every component instead of doing it in a central palce. Looking at the examples there might also be usability aspects behind the way they tied the font size to the button sizes.

Reply:
Hi, I agree if we can avoid these manual configuration per component, if possible. When looking at the Figma you can see we have to use the `fontSize:md` to button size `lg` and `xl`. If using the button size (passed by consumer), I don't see any possibility to implement the `fontSize` and `space` dynamically. Please share the adaptation if you have, I'm happy to modify the code

Reply:
We need to make the buttons more consistent. We can check with the designer to keep only one fontSize for the button. Besides you can use the typography tokens for the text, then you won't need to add the fontWeight in the `commonNormalState` styles

Reply:
I've tried to minimized as much as possible the manual mapping for spacing (regarding padding style) by using pre-defined fixed height and typography. However, we still need to define those mapping for sizes:

```ts
const sizes = {
  md: {
    height: buttonHeights.md,
    px: 'md',
    ...typography.heading5,
  },
  lg: {
    height: buttonHeights.lg,
    px: 'lg',
    ...typography.heading4,
  },
  xl: {
    height: buttonHeights.xl,
    px: 'xl',
    ...typography.heading4,
  },
};
```

If you have a better approach to implement spacing for different sizes, please let me know.